test/distributed/org/apache/cassandra/distributed/test/AbstractNetstatsStreaming.java

```diff
@@ -523,6 +525,20 @@ public NetstatResults call() throws Exception
                         {
                             sawAnyStreamingOutput = true;
                         }
+                        else
+                        {
+                            // there is a race condition that streaming starts/stops between calls to netstats
+                            // to detect this, check to see if the node has completed a stream
+                            // expected log: [Stream (.*)?] All sessions completed
+                            LogResult<List<String>> logs = node.logs().grep(mark, "\\[Stream .*\\] All sessions completed");
+                            mark = logs.getMark();
+                            if (!logs.getResult().isEmpty())
+                            {
+                                // race condition detected...
+                                logger.warn("Test race condition detected where streaming started/stopped between calls to netstats");
```

Comment:
What do you hope a user does in response to this warning? Might be a nit, but I see this as INFO. Seems perfectly fine if a test runner is running slowly and finishes streaming before it can call netstat again, or if the test changes in the future so the streaming happens more quickly.